Doc: Specs gtfs-rt2kirin (#306)
CanalTP/kirin: documentation/gtfsrt_connector.md

# GTFS-RT Connector

## Overview
This document describes how a [GTFS-RT](https://gtfs.org/reference/realtime/v2/) feed is read in Kirin.

Currently, the connector can only read TripUpdate messages with arrival/departure delays (no added/deleted trip/stop).

## Connector description

> For the sake of simplicity, in the following only the input relevant fields are 

Reviewer: It feels like something is missing here, at least a comma and probably a word.

```suggestion
For the sake of simplicity, in the following document, only the input relevant fields are 
```

described, leaving aside the fields of the internal model that are managed by Kirin.

### RealTimeUpdate
Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---
connector |  | Fixed value `gtfs-rt`.
raw_data | _Complete received feed_ |
contributor |  | Fixed value specified in the configuration of Kirin.
trip_updates |  | List of trip updates information, see `TripUpdates` below.

### TripUpdate
Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---

> vj_id | trip_id | The id of the Navitia `VehicleJourney` having a code of type `source` and a value that matches the input trip_id.

Reviewer: I'm not sure it is clear what does the `and` link together in this sentence. Is it `The id (...) and a value (...)`?  Or `a code (...) and a value (...)`?

# GTFS-RT Connector

## Overview
This document describes how a [GTFS-RT](https://gtfs.org/reference/realtime/v2/) feed is read in Kirin.

Currently, the connector can only read TripUpdate messages with arrival/departure delays (no added/deleted trip/stop).

## Connector description
For the sake of simplicity, only the input relevant fields are described below, 
leaving aside the fields of the internal model that are managed by Kirin.

### RealTimeUpdate
Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---
connector |  | Fixed value `gtfs-rt`.

> raw_data | _Complete received feed_ |

Reviewer 2: I think we should add that the stored data in in JSON (not PBF)

contributor |  | Fixed value specified in the configuration of Kirin.
trip_updates |  | List of trip updates information, see `TripUpdates` below.

### TripUpdate
Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---

> vj_id | trip_upadte.trip.trip_id | The id of the Navitia `VehicleJourney` being updated by the `TripUpdate`. See the mapping methode below.

Reviewer 2: typos
```suggestion
vj_id | trip_update.trip.trip_id | The id of the Navitia `VehicleJourney` being updated by the `TripUpdate`. See the mapping method below.
```

> vj_id | trip_update.trip.trip_id (indirectly) | The id of the Navitia `VehicleJourney` being updated by the `TripUpdate`. See the mapping methode below.

Reviewer 3: minor typo left:
```suggestion
vj_id | trip_update.trip.trip_id (indirectly) | The id of the Navitia `VehicleJourney` being updated by the `TripUpdate`. See the mapping method below.
```

> message | ?? | 

Reviewer 2: Is this field used ? (@pbougue ?)

contributor |  | Fixed value specified in the configuration of Kirin.
stop_time_updates |  | List of arrival/departure time updates at stops for this trip, see `StopTimeUpdates` below.
effect |  | Effect is set to `SIGNIFICANT_DELAYS` when the trip status is `update` and `UNKNOWN_EFFECT` otherwise.

### VehicleJourney
The right Navitia trip that is impacted by a given realtime update is retrieved 
through a call to Navitia. The call should search for the `VehicleJourney` that has 
a code of type `source` and a value matching the input trip_id.

The date used to narrow down the research of the right Navitia trip is defined by the date found in `header.timestamp`.

Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---
navitia_trip_id | trip_upadte.trip.trip_id | `trip_id` of the `VehicleJourney` in Navitia. See above for the mapping rule.
start_timestamp | header.timestamp | Start datetime of the `VehicleJourney` in Navitia.

### StopTimeUpdate
Once the right Navitia trip is identified, the timing information for each `stop_time` is updated.

> 

Reviewer 2: Do I understand correctly if I say that when a GTFS-RT is referencing a new trip, the trip is "downloaded" from navitia with all it's stop_times, and then updated with RT feed infos ?
If all the stop_times are stored, I think we should add it to the spec.

Kirin property | GTFS-RT object | Comment/Mapping rule
--- | --- | ---
order |  | `stop_time` order of this stop in the `VehicleJourney`. 
stop_id | trip_update.stop_time_update.stop_id | The id of the updated stop in Navitia that has a code of type `source` and a value matching the input stop_id.
message | ?? | 

> departure |  | Departure datetime of the `VehicleJourney` for this stop in Navitia. 

Reviewer 3: It's the **RT** datetime (and yes, code is really messed up for that):
```suggestion
departure |  | Realtime departure datetime for this stop (including delay). 
```

departure_delay | trip_update.stop_time_update.departure.delay | 

> departure_status |  | Status is set to `none` if the departure delay is 0, otherwise status is set to `updated`.

Reviewer 3: ```suggestion
departure_status |  | Status is set to `none` if the departure delay is 0 or not provided, otherwise status is set to `update`.
```

> arrival |  | Arrival datetime of the `VehicleJourney` for this stop in Navitia.

Reviewer 3: Same:
```suggestion
arrival |  | Realtime arrival datetime for this stop (including delay).
```

arrival_delay | trip_update.stop_time_update.arrival.delay | 

> arrival_status |  | Status is set to `none` if the departure delay is 0, otherwise status is set to `updated`.

Reviewer 3: wrong copy/paste + typo:
```suggestion
arrival_status |  | Status is set to `none` if the arrival delay is 0 or not provided, otherwise status is set to `update`.
```
